ts/parts/Utilities.ts

```diff
@@ -1427,6 +1444,23 @@ function defined<T>(obj: T): obj is NonNullable<T> {
     return typeof obj !== 'undefined' && obj !== null;
 }
 
+function attr(
+    elem: (Highcharts.HTMLDOMElement|Highcharts.SVGDOMElement)
+): undefined;
```

Comment:
I do not see any value in this definition. It is possible, true, but not useful.

Reply:
I agree that it is not useful, and probably a bug if used like this, but this is how it is currently implemented.

If we should remove this overload, then I would prefer to also correct the implementation.

Reply:
Still the compiler should not accept it.

Reply:
I actually do not see any problem with the implementation. You just have to remove the optional flag for the second parameter.

Reply:
> remove the optional flag

Yes, that is a good suggestion, but then we can also optimize the following:
```javascript
} else if (defined(prop) && isObject(prop)) {
```
to:
```javascript
} else {
```
because `prop` is always defined, and if it is not a string then it has to be an object.

So I still believe an adjustment of the implementation is in order.

Reply:
Agreed. The adjustment will cover it.

Reply:
Great, will make the changes ASAP

Reply:
@bre1470  and @TorsteinHonsi please see latest changes.

Reply:
This looks good.